src/runtime/pythonexception.cs

```diff
@@ -148,6 +149,18 @@ public string PythonTypeName
             get { return _pythonTypeName; }
         }
 
+        /// <summary>
+        /// Replaces PyValue with an instance of PyType, if PyValue is not already an instance of PyType.
+        /// Often PyValue is a string and this method will replace it with a proper exception object.
+        /// Must not be called when an error is set.
```

Comment:
Hm. Maybe add a check then?

Also, I can't find that requirement in the docs. Where is it coming from?

Reply:
I found that requirement by experimenting.  I will add a code comment explaining what happens.

Reply:
This is weird. In the Python source code there's no indication, that this function has anything to do with the current error. I guess it is fine to leave the comment though.